solr/solrj/src/java/org/apache/solr/client/solrj/request/HealthCheckRequest.java

```diff
@@ -21,11 +21,17 @@
 import org.apache.solr.client.solrj.SolrRequest;
 import org.apache.solr.client.solrj.impl.HttpSolrClient;
 import org.apache.solr.client.solrj.response.HealthCheckResponse;
+import org.apache.solr.common.params.ModifiableSolrParams;
 import org.apache.solr.common.params.SolrParams;
 
+import java.util.OptionalInt;
+
 import static org.apache.solr.common.params.CommonParams.HEALTH_CHECK_HANDLER_PATH;
 
 public class HealthCheckRequest extends SolrRequest<HealthCheckResponse> {
+    public static final String PARAM_MAX_GENERATION_LAG = "maxGenerationLag";
+
+    private OptionalInt maxLagAllowed = OptionalInt.empty();
```

Comment:
Why are we using this over an Integer that might be null? Or something like -1 meaning infinite lag?

Reply:
I want distinguish between user using invalid values i.e. negative values, not specifying value and specifying a valid value. We aren't really stopping users from using negative values but if they do for instance user -1, they don't have wonder if it the value they used or came via some default. 

I think `Optional` in java I a god way to distinguish between these cases